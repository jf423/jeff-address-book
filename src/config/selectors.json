{
    "INDEX_PAGE": {
        "INDEX_CONTAINER": {
            "ID": "INDEX_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='INDEX_CONTAINER']",
            "EX": "[data-selector-id='INDEX_CONTAINER']",
            "PARAM": ""
        },
        "SEARCH_CONTAINER": {
            "ID": "SEARCH_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_CONTAINER']",
            "EX": "[data-selector-id='SEARCH_CONTAINER']",
            "PARAM": ""
        },
        "SEARCH_INPUT": {
            "ID": "SEARCH_INPUT",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_INPUT']",
            "EX": "[data-selector-id='SEARCH_INPUT']",
            "PARAM": ""
        },
        "SEARCH_CLEAR": {
            "ID": "SEARCH_CLEAR",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_CLEAR']",
            "EX": "[data-selector-id='SEARCH_CLEAR']",
            "PARAM": ""
        },
        "SEARCH_BUTTON": {
            "ID": "SEARCH_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_BUTTON']",
            "EX": "[data-selector-id='SEARCH_BUTTON']",
            "PARAM": ""
        },
        "USER_CONTAINER": {
            "ID": "USER_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CONTAINER']",
            "EX": "[data-selector-id='USER_CONTAINER']",
            "PARAM": ""
        },
        "USER_CONTENT": {
            "ID": "USER_CONTENT",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CONTENT']",
            "EX": "[data-selector-id='USER_CONTENT']",
            "PARAM": ""
        },
        "USER_CATALOG_END": {
            "ID": "USER_CATALOG_END",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CATALOG_END']",
            "EX": "[data-selector-id='USER_CATALOG_END']",
            "PARAM": ""
        },
        "USER_CELL_CONTAINER": {
            "ID": "USER_CELL_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1']",
            "PARAM": ""
        },
        "USER_CELL_CONTENT": {
            "ID": "USER_CELL_CONTENT",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_CONTENT']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_CONTENT']",
            "PARAM": ""
        },
        "USER_CELL_AVATAR": {
            "ID": "USER_CELL_AVATAR",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_AVATAR']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_AVATAR']",
            "PARAM": ""
        },
        "USER_CELL_NAME": {
            "ID": "USER_CELL_NAME",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_NAME']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_NAME']",
            "PARAM": ""
        },
        "USER_CELL_NATIONALITY": {
            "ID": "USER_CELL_NATIONALITY",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_NATIONALITY']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_NATIONALITY']",
            "PARAM": ""
        },
        "USER_CELL_USERNAME": {
            "ID": "USER_CELL_USERNAME",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_USERNAME']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_USERNAME']",
            "PARAM": ""
        },
        "USER_CELL_EMAIL": {
            "ID": "USER_CELL_EMAIL",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_EMAIL']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_EMAIL']",
            "PARAM": ""
        },
        "USER_MODAL_CONTAINER": {
            "ID": "USER_MODAL_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CONTAINER']",
            "EX": "[data-selector-id='USER_MODAL_CONTAINER']",
            "PARAM": ""
        },
        "USER_MODAL_CLOSE_BUTTON": {
            "ID": "USER_MODAL_CLOSE_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CLOSE_BUTTON']",
            "EX": "[data-selector-id='USER_MODAL_CLOSE_BUTTON']",
            "PARAM": ""
        },
        "USER_MODAL_AVATAR": {
            "ID": "USER_MODAL_AVATAR",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_AVATAR']",
            "EX": "[data-selector-id='USER_MODAL_AVATAR']",
            "PARAM": ""
        },
        "USER_MODAL_NAME": {
            "ID": "USER_MODAL_NAME",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_NAME']",
            "EX": "[data-selector-id='USER_MODAL_NAME']",
            "PARAM": ""
        },
        "USER_MODAL_STREET": {
            "ID": "USER_MODAL_STREET",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_STREET']",
            "EX": "[data-selector-id='USER_MODAL_STREET']",
            "PARAM": ""
        },
        "USER_MODAL_CITY": {
            "ID": "USER_MODAL_CITY",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CITY']",
            "EX": "[data-selector-id='USER_MODAL_CITY']",
            "PARAM": ""
        },
        "USER_MODAL_STATE": {
            "ID": "USER_MODAL_STATE",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_STATE']",
            "EX": "[data-selector-id='USER_MODAL_STATE']",
            "PARAM": ""
        },
        "USER_MODAL_POSTCODE": {
            "ID": "USER_MODAL_POSTCODE",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_POSTCODE']",
            "EX": "[data-selector-id='USER_MODAL_POSTCODE']",
            "PARAM": ""
        },
        "USER_MODAL_PHONE": {
            "ID": "USER_MODAL_PHONE",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_PHONE']",
            "EX": "[data-selector-id='USER_MODAL_PHONE']",
            "PARAM": ""
        },
        "USER_MODAL_CELL": {
            "ID": "USER_MODAL_CELL",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CELL']",
            "EX": "[data-selector-id='USER_MODAL_CELL']",
            "PARAM": ""
        }
    },
    "SETTINGS_PAGE": {
        "SETTINGS_CONTAINER": {
            "ID": "SETTINGS_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='SETTINGS_CONTAINER']",
            "EX": "[data-selector-id='SETTINGS_CONTAINER']",
            "PARAM": ""
        },
        "NATIONALITIES_CONTAINER": {
            "ID": "NATIONALITIES_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CONTAINER']",
            "EX": "[data-selector-id='NATIONALITIES_CONTAINER']",
            "PARAM": ""
        },
        "NATIONALITIES_BACK_BUTTON": {
            "ID": "NATIONALITIES_BACK_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_BACK_BUTTON']",
            "EX": "[data-selector-id='NATIONALITIES_BACK_BUTTON']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_CH": {
            "ID": "NATIONALITIES_CHECKBOX_CH",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_CH']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_CH']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_ES": {
            "ID": "NATIONALITIES_CHECKBOX_ES",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_ES']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_ES']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_FR": {
            "ID": "NATIONALITIES_CHECKBOX_FR",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_FR']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_FR']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_GB": {
            "ID": "NATIONALITIES_CHECKBOX_GB",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_GB']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_GB']",
            "PARAM": ""
        },
        "NATIONALITIES_SAVE_BUTTON": {
            "ID": "NATIONALITIES_SAVE_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_SAVE_BUTTON']",
            "EX": "[data-selector-id='NATIONALITIES_SAVE_BUTTON']",
            "PARAM": ""
        }
    }
}