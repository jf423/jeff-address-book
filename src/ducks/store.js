import { useMemo } from 'react';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer, { initialState } from './modules/users';

let store;

export function initStore(preloadedState = initialState) {
    return createStore(
        reducer,
        preloadedState,
        composeWithDevTools(applyMiddleware())
    );
}

export function initializeStore(preloadedState) {
    const _store = store || initStore(preloadedState);
    if (typeof window === 'undefined') return _store;
    if (!store) store = _store;

    return _store;
}

export function useStore(state) {
    return useMemo(() => initializeStore(state), [state]);
}
