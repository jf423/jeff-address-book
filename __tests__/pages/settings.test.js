import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { toBeInTheDocument } from '@testing-library/jest-dom/matchers';

import { initStore } from '@/ducks/store.js';

import { SETTINGS_PAGE } from '@/config/selectors.json';

import Settings from '../../pages/settings.js';

const mockDispatch = jest.fn();

expect.extend({ toBeInTheDocument });

describe('Settings component test', () => {
    let settings;
    const store = initStore();
    store.dispatch = mockDispatch;
    const renderSettings = props => (
        <Provider store={store}>
            <Settings {...props} />
        </Provider>
    );

    beforeEach(() => {
        settings = render(renderSettings());
    });

    afterEach(() => {
        settings.unmount();
    });

    test('should display as expected', () => {
        const settingsContainer = settings.getByTestId(SETTINGS_PAGE.SETTINGS_CONTAINER.ID);

        expect(settingsContainer).toBeInTheDocument();
        expect(settings).toMatchSnapshot();
    });
});
