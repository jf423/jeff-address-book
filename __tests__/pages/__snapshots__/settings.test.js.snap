// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component test should display as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="settings__SettingsContainer-j34y83-0 cLbnm"
        data-selector-id="SETTINGS_CONTAINER"
      >
        <div
          class="Nationalities__Container-zzgmgu-0 gMhHae"
          data-selector-id="NATIONALITIES_CONTAINER"
        >
          <div
            class="Nationalities__Header-zzgmgu-1 UTaLr"
          >
            <button
              aria-label="back"
              class="MuiButtonBase-root MuiIconButton-root"
              data-selector-id="NATIONALITIES_BACK_BUTTON"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="Nationalities__SettingsContent-zzgmgu-2 iMsDgK"
          >
            <div
              class="MuiFormGroup-root"
            >
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                  data-selector-id="NATIONALITIES_CHECKBOX_CH"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input-4"
                      data-indeterminate="false"
                      name="ch"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Switzerland (CH)
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                  data-selector-id="NATIONALITIES_CHECKBOX_ES"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input-4"
                      data-indeterminate="false"
                      name="es"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Spain (ES)
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                  data-selector-id="NATIONALITIES_CHECKBOX_FR"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input-4"
                      data-indeterminate="false"
                      name="fr"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  France (FR)
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                  data-selector-id="NATIONALITIES_CHECKBOX_GB"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input-4"
                      data-indeterminate="false"
                      name="gb"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  United Kingdom (GB)
                </span>
              </label>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-selector-id="NATIONALITIES_SAVE_BUTTON"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Save
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="settings__SettingsContainer-j34y83-0 cLbnm"
      data-selector-id="SETTINGS_CONTAINER"
    >
      <div
        class="Nationalities__Container-zzgmgu-0 gMhHae"
        data-selector-id="NATIONALITIES_CONTAINER"
      >
        <div
          class="Nationalities__Header-zzgmgu-1 UTaLr"
        >
          <button
            aria-label="back"
            class="MuiButtonBase-root MuiIconButton-root"
            data-selector-id="NATIONALITIES_BACK_BUTTON"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="Nationalities__SettingsContent-zzgmgu-2 iMsDgK"
        >
          <div
            class="MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                data-selector-id="NATIONALITIES_CHECKBOX_CH"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-4"
                    data-indeterminate="false"
                    name="ch"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Switzerland (CH)
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                data-selector-id="NATIONALITIES_CHECKBOX_ES"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-4"
                    data-indeterminate="false"
                    name="es"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Spain (ES)
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                data-selector-id="NATIONALITIES_CHECKBOX_FR"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-4"
                    data-indeterminate="false"
                    name="fr"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                France (FR)
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                data-selector-id="NATIONALITIES_CHECKBOX_GB"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-4"
                    data-indeterminate="false"
                    name="gb"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                United Kingdom (GB)
              </span>
            </label>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-selector-id="NATIONALITIES_SAVE_BUTTON"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Save
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
